app/services/ExerciseService.ts

```diff
@@ -94,26 +97,73 @@ export const ExerciseService = {
   },
 
   /**
-* Fetches content and metadata
-*
-* @param {string} exerciseID id for an exercise
-* @return {JSON} the JSON being returned
-* @throws {NotFoundException} File is Not found exception handler
-*/
-  fetchData(exerciseID: string) {
-    try {
-      var content = this.getContent(exerciseID)
-      var metadata = this.getMetaData(exerciseID)
-      var display = {};
+   * gets status of the problem
```

Comment:
Instead of problem, try to use word exercise to be more consistent.
Also from here and down, create return type of each function.